lib/pages/share/share.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:collection/collection.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/usecase/send_file_interactor.dart';
+import 'package:fluffychat/domain/usecase/send_image_interactor.dart';
+import 'package:fluffychat/pages/share/share_view.dart';
+import 'package:fluffychat/presentation/extensions/send_file_extension.dart';
+import 'package:fluffychat/presentation/mixins/send_files_mixin.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/matrix_file_extension.dart';
+import 'package:fluffychat/widgets/matrix.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:matrix/matrix.dart';
+import 'package:scroll_to_index/scroll_to_index.dart';
+
+class Share extends StatefulWidget {
+  const Share({super.key});
+
+  @override
+  State<Share> createState() => ShareController();
+}
+
+class ShareController extends State<Share> with SendFilesMixin {
+  final sendFileInteractor = getIt.get<SendFileInteractor>();
+
+  final sendImagesInteractor = getIt.get<SendImageInteractor>();
+
+  final isShowRecentlyChatsNotifier = ValueNotifier(true);
+
+  final AutoScrollController recentChatScrollController =
+      AutoScrollController();
+
+  final selectedRoomsNotifier = ValueNotifier(<String>[]);
+
+  void onSelectChat(String id) {
+    if (selectedRoomsNotifier.value.contains(id)) {
+      selectedRoomsNotifier.value.remove(id);
+    } else {
+      selectedRoomsNotifier.value.add(id);
+    }
+    selectedRoomsNotifier.value = selectedRoomsNotifier.value.sorted(
+      (current, next) => current.compareTo(next),
+    );
```

Comment:
it will be rebuilt twice
clone a value, modify it, then set it back to the notifier in the last line

Reply:
i set the `selectedRoomsNotifier.value` only one, so it will not be rebuild twice